docs: add GitProxy project health report H2 2026 (#299)
finos/technical-oversight-committee: project-reports/2026/2026-H2-Git-Proxy.md

| --------------- | ------------- | ---------------------------------- | --------------------------- |
| @06kellyjac     | Jack Kelly    | @controlplaneio                    | jack+finos@control-plane.io |
| @andypols       | Andy Pols     | Qube Research & Technologies (QRT) | andrew.pols@qube-rt.com     |
| @coopernetes    | Thomas Cooper | @RBC                               | thomas.cooper@rbc.com       |
| @grovesy        | Paul Groves   | Citi                               | paul.groves@citi.com        |
| @jescalada      | Juan Escalada | G-Research Open Source Software    | juan.escalada@gr-oss.io     |
| @kriswest       | Kris West     | NatWest Group                      | kristopher.west@natwest.com |

**Repository:** https://github.com/finos/git-proxy

# Project Overview

GitProxy is an application that stands between developers and a Git remote endpoint (e.g., github.com). It applies rules and workflows (configurable as plugins) to all outgoing git push operations to ensure they are compliant.

## Goals for H2 2026
- Increase contributor/contribution counts
- Speed up PR reviews and merge speeds

> - Release GitProxy 2.1 and 2.2, including SSH support and brand-new UI

Reviewer: ```suggestion
- Release a series of minor and patch releases, including new features such as SSH support, a brand-new UI, live-feedback to the end user on scan progress and a host of other small quality-of-life improvements for Git Proxy users and developers
```

# Current Status

## Key accomplishments

Source: [LFX Insights](https://insights.linuxfoundation.org/project/git-proxy/repository/finos_git-proxy/contributors)
- Consistently achieved 10+ participants in our bi-weekly Community Calls
- Increased contributor count from 15/16 in Q3/Q4 '25 to 26 in Q2 '26
- Increased star count by 26% from 193 to 245

## Major milestones achieved

> - Released GitProxy v2 with many improvements including: support beyond GitHub, better handling of fork-based workflows, improved performance for large repositories, clearer documentation and better user experience

Reviewer: ```suggestion
- Released GitProxy v2 with many improvements including: support beyond GitHub, better handling of fork-based workflows, improved performance for large repositories, a full rewrite of git pushes, clearer documentation and better user experience, a wholesale refactor of the codebase to modern typescript
```

## Community or adoption barriers

We're currently working on marketing GitProxy along with FINOS in hopes of getting new contributors and future maintainers onboard. It's not entirely clear what the target audience should be, so we would appreciate any examples of successful marketing campaigns for other projects. 

# Roadmap & Goals for Next 6 Months
- Goal: Onboard 1 new actively contributing organization (financial firm or otherwise)
  - By "actively contributing", we mean active in making issues, PRs and reviews, enough to elect a new maintainer from that organization
  - An organization may assign multiple developers to the project
- Upcoming milestones/releases
  - GitProxy 2.1 (SSH Release + various other features)
  - GitProxy 2.2 (Full UI revamp)
- Planned features
  - Porting various QoL features from [fogwall](https://github.com/RBC/fogwall) including:
    - Real-time feedback on pushes https://github.com/finos/git-proxy/pull/1637
    - Deferred forwarding of approved pushes (for long-running checks)
    - In-process credential handling to skip double-push requirement
    - Client disconnect detection

# TOC Support Needed

> As GitProxy matures, we find ourselves needing more people to help us review PRs and make minor improvements that maintainers often don't have the time to do. Therefore, we'd really appreciate any support in getting GitProxy out in the world.

Reviewer: You could mention your PR triage AI integration here

Author: Good catch!

We're also actively marketing the project, and open to collaborating with other FINOS or LF projects that could benefit from including GitProxy as a dependency or to solve a specific problem.

# Additional Information

> @coopernetes recently contributed [Fogwall](https://github.com/RBC/fogwall), a Java reimplementation of GitProxy that [solves some pain points](https://gist.github.com/coopernetes/d02d48efa759282ff8187da0d5dcae64) specific to his organization. We're working on [porting some of those features](https://github.com/finos/git-proxy/issues/1638) into GitProxy.

Reviewer 2: This project has not been contributed to FINOS and exists as a separate, RBC FOSS project (for now). Just want to be accurate of the current state.
```suggestion
@coopernetes recently created [fogwall](https://github.com/RBC/fogwall), a Java reimplementation of GitProxy that [solves some pain points](https://gist.github.com/coopernetes/d02d48efa759282ff8187da0d5dcae64) specific to his organization. We're working on [porting some of those features](https://github.com/finos/git-proxy/issues/1638) into GitProxy.
```
